src/editor.c

```diff
@@ -5172,6 +5172,9 @@ void editor_apply_update_prefs(GeanyEditor *editor)
 	/* virtual space */
 	SSM(sci, SCI_SETVIRTUALSPACEOPTIONS, editor_prefs.show_virtual_space, 0);
 
+	/* input method editor's candidate window behaviour */
+	SSM(sci, SCI_SETIMEINTERACTION, editor_prefs.ime_interaction, 0);
+
```

Comment:
The current situation is that no IME interaction is set, unless you can be sure that setting an interaction has no effect anywhere else, then not setting it at all should be the default.  

So you can use your boolean and put this in an if.  That would also actually match your documentation above.